### ISSUE TITLE
Unable to use IPSec encryption algorithms other than rfc4106(gcm(aes))

### ISSUE THREAD

**Issue by neerajdixit-msft2** [cilium/cilium#28318]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

We are using cilium with IPSec encryption for K8s cluster deployment. Based on the project's recommendation, we wanted to try out different encryption algorithms like **gcm(aes)** instead of the documented **rfc4106(gcm(aes))** - https://docs.cilium.io/en/stable/security/network/encryption-ipsec/#generate-import-the-psk

But when its deployed, I get an error stating that it's an invalid algorithm name!
`2023-09-27T12:48:52.909925469-07:00 level=info msg="Loading IPsec keyfile" file-path=/etc/ipsec/keys subsys=ipsec`
`2023-09-27T12:48:52.909968995-07:00 level=error msg="Failed to load IPsec keyfile" error="invalid AEAD algorithm \"gcm(aes)\"" subsys=ipsec`

On checking the code, I found out that all other algorithms are explicitly blocked - https://github.com/cilium/cilium/blob/9ab742b6ed2c483d75b49292d250258a771b3b31/pkg/datapath/linux/ipsec/ipsec_linux.go#L824C1-L827C5

Can we get a list of supported encryption algorithms for cilium? Is it already documented elsewhere?

All your help is appreciated.

### Cilium Version

1.14.2

### Kernel Version

5.15.126.1-1.cm2 #1

### Kubernetes Version

1.25.5

### Relevant log output

```shell
2023-09-27T12:48:52.909925469-07:00 level=info msg="Loading IPsec keyfile" file-path=/etc/ipsec/keys subsys=ipsec
2023-09-27T12:48:52.909968995-07:00 level=error msg="Failed to load IPsec keyfile" error="invalid AEAD algorithm \"gcm(aes)\"" subsys=ipsec
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by soggiest** [comment]
I happened to need the answer to a very similar question yesterday and found the info under `ALGO-LIST` in the [IP-XFRM manual](https://man7.org/linux/man-pages/man8/ip-xfrm.8.html) was useful for understanding exactly which algorithms are supported. Based on that doc you could try out `rfc4543(gcm(aes))` as a different GCM-AEAD algorithm.

I hope this helps!

**Comment by neerajdixit-msft2** [comment]
Thank you, that helps.

The reason for me to check for new AEAD implementations was to enable 256-bit encryption while using IPSec.
I am using RFC 4543 implementation since it supports 256 bit encryption - https://datatracker.ietf.org/doc/html/rfc4543#section-3.4
But even when using the **rfc4543(gcm(aes))** implementation, I am still unable to use a 256-bit key:
`2023-10-02T11:45:22.021703235-07:00 level=error msg="IPsec enable failed" error="unable to replace remote state: function not implemented" local-ip=0.0.0.0/0 reason="out IPv4" remote-ip=10.0.0.0/24 spi=0 subsys=linux-datapath`

I am using the **cilium-ipsec-keys** as follows:
`3 rfc4543(gcm(aes)) <PSK> 256`

Am I missing anything here?

-Neeraj

**Comment by neerajdixit-msft2** [comment]
Actually, let me create a new issue since this is a different question. Thank you for all the help.
